packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -30,24 +29,29 @@ export default function createCssVarsProvider(options) {
     resolveTheme,
   } = options;
 
+  const defaultContext = {
+    allColorSchemes: [],
+    colorScheme: undefined,
+    darkColorScheme: undefined,
+    lightColorScheme: undefined,
+    mode: undefined,
+    setColorScheme: () => {},
+    setMode: () => {},
+    systemMode: undefined,
+  };
+
   const ColorSchemeContext = React.createContext(undefined);
 
   if (process.env.NODE_ENV !== 'production') {
     ColorSchemeContext.displayName = 'ColorSchemeContext';
   }
 
-  const useColorScheme = () => {
-    const value = React.useContext(ColorSchemeContext);
-    if (!value) {
-      throw new MuiError('MUI: `useColorScheme` must be called under <CssVarsProvider />');
-    }
-    return value;
-  };
+  const useColorScheme = () => React.useContext(ColorSchemeContext) || defaultContext;
```

Comment:
It's no longer make sense to throw error mentioning `CssVarsProvider`. It should be callable anywhere similar to `useTheme`.

This allow apps like [Toolpad](https://github.com/mui/mui-toolpad/pull/3872) to have control of the theme structure.